estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,59 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+
+
+class PropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Real Estate offer"
+    _sql_constraints = [
+        ('check_price', 'CHECK(price > 0)',
+         'A property offer must be strictly positive.'),
+    ]
+
+    price = fields.Float()
+    status = fields.Selection(
+        copy=False, selection=[("accepted", "Accepted"), ("refused", "Refused")]
+    )
+    partner_id = fields.Many2one("res.partner", required=True, string="Buyer")
+    property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(default=7, string="Validity (days)")
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline", inverse="_compute_validity", string="Deadline"
+    )       
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
```

Comment:
```suggestion
        for offer in self:
```
naming your variable `record` when iterating over `self` is fine, but we usually try to be more explicit. When working with lots of models and long files, it makes things easier.